sstmac/hardware/snappr/snappr_switch.cc

```diff
@@ -176,17 +176,17 @@ SnapprSwitch::SnapprSwitch(uint32_t id, SST::Params& params) :
   TimeDelta byte_delay(1.0/link_bw_);
   std::string sw_arbtype = params.find<std::string>("arbitrator", "fifo");
   std::string link_arbtype = link_params.find<std::string>("arbitrator", sw_arbtype);
-  outports_.reserve(top_->maxNumPorts());
+  outports_.resize(top_->maxNumPorts());
```

Comment:
Optimizer _probably_ fixes this but you should do something like:
```cpp
auto maxPorts = top_->maxNumPorts();
outports_.resize(maxPorts);
inports_.resize(maxPorts);
for(int i = 0; i < maxPorts; ++i){
```
Then you can be sure you don't trigger 3 loads for a single value. Up to you though just a suggestion.

Reply:
Sure, worry about this later. Won't fix for now.